### PR TITLE
SOC-1402 | Add memcache over UserAttributeDao

### PR DESCRIPTION
**Pull request by kvas-damian**
https://wikia-inc.atlassian.net/browse/SOC-1402

### CODE REVIEW

File: service/user-attribute/src/main/java/com/wikia/services/userattribute/daos/MemcachedUserAttributeDao.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import java.util.stream.Collectors;
+
+import javax.inject.Named;
+
+public class MemcachedUserAttributeDao implements UserAttributeDaoInterface {
+
+  private MemcachedClient memcachedClient;
+  private final int CACHE_TTL = 24 * 60 * 60;
+  private static final Logger LOGGER = LoggerFactory.getLogger(MemcachedUserAttributeDao.class);
+  private UserAttributeDAO userAttrbiuteDao;
```
Comment (bekcunning) [inline service/user-attribute/src/main/java/com/wikia/services/userattribute/daos/MemcachedUserAttributeDao.java]: Attribute is misspelled in your variable name here.

File: service/user-attribute/src/main/java/com/wikia/services/userattribute/daos/MemcachedUserAttributeDao.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.inject.Named;
+
+public class MemcachedUserAttributeDao implements UserAttributeDaoInterface {
+
+  private MemcachedClient memcachedClient;
+  private final int CACHE_TTL = 24 * 60 * 60;
```
Comment (WikiaTech) [inline service/user-attribute/src/main/java/com/wikia/services/userattribute/daos/MemcachedUserAttributeDao.java]: MAJOR Rename this field "CACHE_TTL" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00116)
MAJOR Make this final field static too. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1170)

File: service/user-attribute/src/main/java/com/wikia/services/userattribute/daos/MemcachedUserAttributeDao.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+  @SuppressWarnings("unchecked")
+  protected List<UserAttribute> getAllAttributesFromCache(String cacheKey) {
+    List<UserAttribute> result = null;
+    Object cacheValue = memcachedClient.get(cacheKey);
+
+    if (cacheValue != null) {
+      try {
+        result = (List<UserAttribute>) cacheValue;
+      } catch (ClassCastException ex) {
```
Comment (WikiaTech) [inline service/user-attribute/src/main/java/com/wikia/services/userattribute/daos/MemcachedUserAttributeDao.java]: MAJOR Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Comment by jsutterfield** [comment]
LGTM. Since we're swapping out our old cacheDoa implementation, can we remove [this file](https://github.com/Wikia/pandora/blob/master/service/user-attribute/src/main/java/com/wikia/services/userattribute/daos/CachedUserAttributeDao.java) too?

File: service/user-attribute/src/main/java/com/wikia/services/userattribute/daos/MemcachedUserAttributeDao.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
```
Comment (WikiaTech) [inline service/user-attribute/src/main/java/com/wikia/services/userattribute/daos/MemcachedUserAttributeDao.java]: MAJOR Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by kvas-damian** [comment]
@jsutterfield removed

**Comment by jsutterfield** [comment]
:+1:

**Pull request merged**